docs/source/developing-volttron/developing-agents/using-asyncio-in-agents.rst

```diff
@@ -0,0 +1,130 @@
+.. _Using-Asyncio-In-Agents:
+
+===================================
+Using Asyncio in Agents
+===================================
+
+The purpose of this section to is to show how to use Asyncio with Gevent within the Agent development framework.
+
+Before we dive into the example, we establish the following concepts:
+
+* An Event Loop is a software design pattern that handles events concurrently; it waits for and dispatches multiple events concurrently and gives the illusion of executing the events in "parallel". In Python, the `Event Loop <https://docs.python.org/3.10/library/asyncio-eventloop.html#event-loop>`_ contains a list of Tasks which controls when and how those Tasks are executed.
+
+* A `Task <https://docs.python.org/3/library/asyncio-task.html#task-object>`_ is an object that runs a `coroutine <https://docs.python.org/3/library/asyncio-task.html#coroutine>`_ (i.e. asynchronous function). In Python, coroutines are written using the 'async' keyword.
+
+* A `Greenlet <https://greenlet.readthedocs.io/en/latest/#instantiation>`_ is a "lightweight coroutine for in-process sequential concurrent programming. Greenlets can be used on their own, but they are frequently used with frameworks such as gevent to provide higher-level abstractions and asynchronous I/O.
+
+* Asyncio is a built-in Python module that allows the developer to write concurrent code.
```

Comment:
* VOLTTRON predates the inclusion of asyncio in python and therefore uses gevent for it's base